src/radix_sort.py

```diff
@@ -0,0 +1,89 @@
+"""Python implemementation of a radix sort algorithm for integers."""
+
+
+def radix_sort(unsorted_list, set_digit=0):
+    """Return sorted list of integers using radix sort algorithm."""
+    if type(unsorted_list) not in [list, tuple] or len(unsorted_list) < 1:
+        raise TypeError('Please enter a list or a tuple with at least one value.')
+    unsorted_list = list(unsorted_list)
+    num_list = [list() for _ in range(10)]
+    sorted_list = []
+    sort_complete = True
+    for item in unsorted_list:
+        _verify_input(item)
+        digit = _get_digit(item, set_digit)
+        if item >= 10 ** (set_digit + 1):
+            sort_complete = False
+        num_list[digit].append(item)
+    for i in range(10):
+        for item in num_list[i]:
+            sorted_list.append(item)
+    if sort_complete:
+        return sorted_list
+    return radix_sort(sorted_list, set_digit=set_digit + 1)
+
+
+def _get_digit(number, place):
+    """Identify target place for a number."""
+    return ((number // 10 ** place) % 10)
```

Comment:
This sometimes returns a float when I run my tests on it creating a typeError. see if you can avoid this.